nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/ReportLineageToAtlas.java

```diff
@@ -584,6 +601,12 @@ public void onTrigger(ReportingContext context) {
             return;
         }
 
+        final String nifiUserId = context.getProperty(NIFI_USER_ID).evaluateAttributeExpressions().getValue();
+        if (isEmpty(nifiUserId)) {
+            getLogger().warn("NiFi user id is empty, make sure '" + NIFI_USER_ID.getDisplayName() + "' is configured correctly.");
+            return;
```

Comment:
Is this for backward compatibility? Reason why I'm asking is, if we are making `NIFI_USER_ID.required(true)` and setting appropriate validator, is this check required?

Reply:
@zenfenan Yes, it is. If NIFI_USER_ID is defined with NiFi Expression Language, evaluation results can be empty if EL refers non-existing properties ... etc.